### PR TITLE
[MCKIN-8511] Optimize - Course cohort page - API

### PR DESCRIPTION
**Pull request by Agrendalath**
@UmanShahzad, this PR is a part of `[MCKIN-8511]` PR-s.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @Agrendalath! I've created [OSPR-2717](https://openedx.atlassian.net/browse/OSPR-2717) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: social_engagement/models.py [hunk 1]
```diff
@@ -124,87 +123,143 @@ def save_user_engagement_score(cls, course_key, user_id, score, stats=None):
...
+        ).exclude(
+            user__in=kwargs.get('exclude_users', [])
+        )
+
+        if kwargs.get('group_ids'):
+            queryset = queryset.filter(user__groups__in=kwargs.get('group_ids')).distinct()
+
+        if kwargs.get('org_ids'):
+            queryset = queryset.filter(user__organizations__in=kwargs.get('org_ids'))
+            queryset = queryset.filter(user__organizations__in=kwargs.get('org_ids'))
```
Comment (UmanShahzad) [inline social_engagement/models.py]: Copy and pasted twice.

**Review by UmanShahzad — COMMENTED** [review]
Thanks, looks good as well.

File: social_engagement/engagement.py [hunk 2]
```diff
@@ -299,7 +299,6 @@ def get_involved_users_in_thread(request, thread):
         _get_details_for_deletion(_get_request(request, params), results=results, is_thread=True)
 
     users = results['users']
-    print(users)
```
[the remark this answers is not in the document]
Comment (Agrendalath) [reply social_engagement/engagement.py:302]: ugh

**Review by UmanShahzad — APPROVED** [review]
:+1:

- [x] I tested this: followed test instructions on the main PR.
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation

**Pull request merged**